extension/entitystore/eksInfo_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"testing"
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+func TestAddPodServiceEnvironmentMapping(t *testing.T) {
+	tests := []struct {
+		name              string
+		want              *ttlcache.Cache[string, ServiceEnvironment]
+		podName           string
+		service           string
+		env               string
+		serviceNameSource string
+		mapNil            bool
+	}{
+		{
+			name: "AddPodWithServiceMapping",
+			want: setupTTLCacheForTesting(map[string]ServiceEnvironment{
+				"test-pod": {
+					ServiceName: "test-service",
+				},
+			}, ttlDuration),
+			podName: "test-pod",
+			service: "test-service",
+		},
+		{
+			name: "AddPodWithServiceEnvMapping",
+			want: setupTTLCacheForTesting(map[string]ServiceEnvironment{
+				"test-pod": {
+					ServiceName: "test-service",
+					Environment: "test-env",
+				},
+			}, ttlDuration),
+			podName: "test-pod",
+			service: "test-service",
+			env:     "test-env",
+		},
+		{
+			name: "AddPodWithServiceEnvMapping",
+			want: setupTTLCacheForTesting(map[string]ServiceEnvironment{
+				"test-pod": {
+					ServiceName:       "test-service",
+					Environment:       "test-env",
+					ServiceNameSource: ServiceNameSourceInstrumentation,
+				},
+			}, ttlDuration),
+			podName:           "test-pod",
+			service:           "test-service",
+			env:               "test-env",
+			serviceNameSource: "Instrumentation",
+		},
+		{
+			name:   "AddWhenPodToServiceMapIsNil",
+			mapNil: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			logger, _ := zap.NewDevelopment()
+			ei := newEKSInfo(logger)
+			if tt.mapNil {
+				ei.podToServiceEnvMap = nil
+			}
+			ei.AddPodServiceEnvironmentMapping(tt.podName, tt.service, tt.env, tt.serviceNameSource)
+			if tt.mapNil {
+				assert.Nil(t, ei.podToServiceEnvMap)
+			} else {
+				for pod, se := range tt.want.Items() {
+					assert.Equal(t, se.Value(), ei.GetPodServiceEnvironmentMapping().Get(pod).Value())
+				}
+				assert.Equal(t, tt.want.Len(), ei.GetPodServiceEnvironmentMapping().Len())
+			}
+		})
+	}
+}
+
+func TestAddPodServiceEnvironmentMapping_TtlRefresh(t *testing.T) {
+	logger, _ := zap.NewDevelopment()
```

Comment:
nit: Can use `zap.NewNop()` for testing.

Reply:
Non-blocking